lib/irb/type_completion/completor.rb

```diff
@@ -0,0 +1,238 @@
+          target + _1[name.size..]
+        end
+      rescue SyntaxError, StandardError => e
+        handle_error(e)
+        []
+      ensure
+        $VERBOSE = verbose
+      end
+
+      def doc_namespace(preposing, matched, postposing, bind:)
+        name = matched[/[a-zA-Z_0-9]*[!?=]?\z/]
+        method_doc = -> type do
+          type = type.types.find { _1.all_methods.include? name.to_sym }
+          if type.is_a? Types::SingletonType
+            "#{Types.class_name_of(type.module_or_class)}.#{name}"
+          elsif type.is_a? Types::InstanceType
+            "#{Types.class_name_of(type.klass)}##{name}"
+          end
+        end
+        call_or_const_doc = -> type do
+          if name =~ /\A[A-Z]/
+            type = type.types.grep(Types::SingletonType).find { _1.module_or_class.const_defined?(name) }
+            type.module_or_class == Object ? name : "#{Types.class_name_of(type.module_or_class)}::#{name}" if type
+          else
+            method_doc.call(type)
+          end
+        end
+
+        value_doc = -> type do
+          return unless type
+          type.types.each do |t|
+            case t
+            when Types::SingletonType
+              return Types.class_name_of(t.module_or_class)
+            when Types::InstanceType
+              return Types.class_name_of(t.klass)
+            when Types::ProcType
+              return 'Proc'
+            end
+          end
+          nil
+        end
+
+        case @result
+        in [:call_or_const, type, _name, _self_call]
+          call_or_const_doc.call type
+        in [:const, type, _name, scope]
+          if type
+            call_or_const_doc.call type
+          else
+            value_doc.call scope[name]
+          end
+        in [:gvar, _name, scope]
+          value_doc.call scope["$#{name}"]
+        in [:ivar, _name, scope]
+          value_doc.call scope["@#{name}"]
+        in [:cvar, _name, scope]
+          value_doc.call scope["@@#{name}"]
+        in [:call, type, _name, _self_call]
+          method_doc.call type
+        in [:lvar_or_method, _name, scope]
+          if scope.local_variables.include?(name)
+            value_doc.call scope[name]
+          else
+            method_doc.call scope.self_type
+          end
+        else
+        end
+      end
+
+      def candidates_from_result(result)
+        candidates = case result
+        in [:require | :require_relative => method, name]
+          if IRB.const_defined? :RegexpCompletor # IRB::VERSION >= 1.8.2
+            path_completor = IRB::RegexpCompletor.new
+          elsif IRB.const_defined? :InputCompletor # IRB::VERSION <= 1.8.1
+            path_completor = IRB::InputCompletor
+          end
+          if !path_completor
+            []
+          elsif method == :require
+            path_completor.retrieve_files_to_require_from_load_path
+          else
+            path_completor.retrieve_files_to_require_relative_from_current_dir
```

Comment:
These could simply be `retrieve_files_to_require_from_load_path` as it inherits `BaseCompletor`?